pystruct/learners/ssvm.py

```diff
@@ -34,12 +36,21 @@ def predict(self, X):
         """
         verbose = max(0, self.verbose - 3)
         if self.n_jobs != 1:
-            prediction = Parallel(n_jobs=self.n_jobs, verbose=verbose)(
-                delayed(inference)(self.model, x, self.w) for x in X)
+            if constraints:
+                prediction = Parallel(n_jobs=self.n_jobs, verbose=verbose)(
+                    delayed(inference)(self.model, x, self.w, constraints=c) for x,c in zip(X, constraints))
+            else:
+                prediction = Parallel(n_jobs=self.n_jobs, verbose=verbose)(
+                    delayed(inference)(self.model, x, self.w) for x in X)
             return prediction
         else:
             if hasattr(self.model, 'batch_inference'):
-                return self.model.batch_inference(X, self.w)
+                if constraints: 
+                    return self.model.batch_inference(X, self.w, constraints=constraints)
```

Comment:
can't the constraints be part of X?

Reply:
Yes...
But constraints can only appear in the X of predict (not in the train) method, at least for now, so keeping them outside the X makes sense as well.
...and now, I'm reluctant to update the code and my other Transkribus code that uses it...
If in the future, constraints get used at train time, then we could then move them in the X.